### ISSUE TITLE
Not able to see which room is in which space during filtering

### ISSUE THREAD

**Issue by daniellekirkwood** [element-hq/element-web#19349]
#### What would you like to do?
When I search for General in Filter, I have a long list (because I'm a part of several Spaces). I need to know which room belongs to which Space so I that I know where to click...

![Screenshot 2021-10-11 at 15 01 59](https://user-images.githubusercontent.com/89144281/136803553-21bc136f-7df5-47a5-929b-d708ffc25ef8.png)
When I search for General in other parts of the product, a si
![Screenshot 2021-10-11 at 14 47 41](https://user-images.githubusercontent.com/89144281/136803534-84673c81-f7ad-4a4d-bc40-e519776113a8.png)
milar thing happens. For example, searching for General when Forwarding a message. I'd suggest adding Grey text after General that shows the Space name.

This is a web version of an android issue: https://github.com/vector-im/element-android/issues/3943

### Have you considered any alternatives?

Design to consider alternatives

**Comment by kittykat** [comment]
See [here](https://github.com/vector-im/element-android/issues/3943) for similar Android issue. Solution should be similar on all three platforms.

**Comment by lilyball** [comment]
Please make sure any such fix also applies without a filter when viewing Home with Show All Rooms checked. I reported this in #19851 but just closed that as a duplicate of this issue.

**Comment by t3chguy** [comment]
1) When user selects the preference of all rooms default to separating by space

+ This is technically feasible but difficult
+ Rooms can be in many spaces so a single room may now be visible and even "active" multiple times on one screen
+ Makes the sorts less useful as now your unread room may be in one of your last spaces and you won't see it off-screen

> and it doesn't work super nicely when a room is in multiple spaces (but that could be solved).

What do you have in mind for this?

> 2) Alternative 1: Use message preview

+ This is technically feasible and easy

> 3) Alternative 2: Tooltip

+ This is technically feasible and easy

---

> Forwarding a message
> Adding existing rooms to space

These two are trivial

**Comment by niquewoodhouse** [comment]
> Please make sure any such fix also applies without a filter when viewing Home with Show All Rooms checked. I reported this in #19851 but just closed that as a duplicate of this issue.

Hi, would you mind explaining a bit more about when and why knowing the Space name is helpful please?  We've looked at some options (above) of where to put it and it would help to have a bit more empathy for you in knowing when in your flow that would be helpful/satisfactory?

Eg - I don't want to hide the message preview feature to enable it in the current design, so maybe the tooltip option works alright for you, but I'm not sure what's the most important?

Thanks for your help

**Comment by lilyball** [comment]
@niquewoodhouse 

It would be helpful when I have rooms of similar names from different spaces favorited. This also applies to the "Favorites" sidebar entry if enabled.

Here's what my Favorites tab looks like right now:

<img width="354" alt="Favorites. There are two rooms. Each room name starts with an emoji and is followed by the word 'General'. This means they look virtually identical in the list." src="https://user-images.githubusercontent.com/714/156079374-4e9789e4-1929-475d-b17e-b44f1e81c9ed.png">

Even without the nearly-duplicated names though, having just the room names without the space names is rather confusing as it requires me to remember what space each room is in, or whether the room is even in a space at all. For example, here's what my Home looks like with Show all rooms enabled:

<img width="353" alt="A list of rooms of various name, like Now Playing, Announcements, Meta, etc. All the rooms are lumped together in a single list" src="https://user-images.githubusercontent.com/714/156079675-f942d6a5-78f1-4e1c-97ec-436d07fe17d7.png">

Which rooms are in spaces? Which space is the room in? The only room that's immediately obvious is "Nix Community" and that's the one room in this list that's not actually a member of a space.

---

My first inclination is to group by space, but I do worry that this would prevent the ability to curate the presentation here (e.g. reordering the rooms), which is a big issue I have with Discord. I think perhaps Home and Favorites should both have an _option_ to "Group by space", and if that's unchecked then it should show the space some other way. In the mockups above I do see a "message preview" thing as the subtitle, which is where I was originally thinking the space would go, but I don't actually see this as a feature in the current client so I'm confused.

**Comment by t3chguy** [comment]
> which is a big issue I have with Discord.

+ 1000

**Comment by gsouquet** [comment]
@niquewoodhouse would you be able to revisit the different alternatives you suggested and propose a way forward.
I personally think that message previews are important but don't think I have useful suggestions to make in this specific scenario

**Comment by niquewoodhouse** [comment]
> @niquewoodhouse would you be able to revisit the different alternatives you suggested and propose a way forward.
> I personally think that message previews are important but don't think I have useful suggestions to make in this specific scenario

Hi @gsouquet I've revisited and my first thought is using the tooltip to say which space it's in is the safest of these options to at least try.  It doesn't solve the underlying problem of seeing at a glance which room is in which space but it's the most straightforward to execute, with our current UI constraints.

![](https://user-images.githubusercontent.com/1587166/155496818-34098be4-e200-44d1-b4ee-f86a13a3ff0c.png)

**Comment by t3chguy** [comment]
@daniellekirkwood this issue seems to be about filtering in the room list which is going away, does spotlight fix your issue sufficiently?

**Comment by shoqvalue** [comment]
I am still having a hell of a time telling rooms apart whenever they are showing without their spaces for grouping.  I know this is complicated, but I haven't had a single user yet who didn't mention it in their first week (as I did). It's definitely a common pain point.  And it's been solved in one dialog somewhere, but I can't now re-find that exception now.  It was something like:

**RoomName  —  "spaceA, spaceB, others."**

And that worked fine. So well, I looked for it in other dialogs, and was sad to not find it. Can't that  be done in any list of rooms? If the list is long, a tooltip could be used for big space lists.
